ports/rp2/machine_pwm.c

```diff
@@ -207,7 +299,7 @@ STATIC void mp_machine_pwm_duty_set_ns(machine_pwm_obj_t *self, mp_int_t duty_ns
         mp_raise_ValueError(MP_ERROR_TEXT("duty larger than period"));
```

Comment:
Not part of your changes, but as long as you are going over things, this error message should change.  It's not checking it against the period (given by TOP), just against the maximum possible value of 65535.

Also the maximum value depends unpredictably on internal details of the PWM implementation.  To make it more predictable and consistent across slight implementation changes, I think we should just cap the value instead of raising an error here.

```python
MicroPython v1.20.0-31-g3e565deee on 2023-04-28; Raspberry Pi Pico with RP2040

Type "help()" for more information.

>>> from machine import PWM, Pin
p=PWM(Pin(25))
p.freq(1000000)
>>> p.duty_ns(1000) # 100% duty
>>> p.duty_ns(500000) # 50,000% duty
>>> p.duty_ns(600000) # 60,000% duty
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: duty larger than period
>>> 
```

Reply:
This error has been removed.